### PR TITLE
PluginList - Refresh button

### PR DESCRIPTION
**Pull request by botatooo**
Add a button in plugin settings to refresh plugins because if you manually add a plugin in the plugins folder it doesn't automatically detect it.

### CODE REVIEW

File: backend/main.py [hunk 1]
```diff
@@ -51,6 +51,7 @@ def __init__(self) -> None:
         if CONFIG["chown_plugin_path"] == True:
             self.web_app.on_startup.append(chown_plugin_dir)
         self.loop.create_task(self.loader_reinjector())
+        self.loop.create_task(self.plugin_reloader())
```
Comment (AAGaming00) [inline backend/main.py:54]: All of this should be rewritten to use inotify, like the plugin hot-reload feature does

**Review by AAGaming00 — CHANGES_REQUESTED** [review]
See comments.

File: frontend/src/plugin-loader.tsx [hunk 2]
```diff
@@ -97,6 +97,16 @@ class PluginLoader extends Logger {
     );
   }
 
+  public refreshPlugins() {
+    this.log('Refreshing plugin list');
+    this.dismountAll();
+    for (const plugin of this.plugins) {
+      this.unloadPlugin(plugin.name);
+    }
+    this.callServerMethod('refresh_plugins');
```
Comment (AAGaming00) [inline frontend/src/plugin-loader.tsx:106]: pretty sure this method should be async and you should await callServerMethod.

**Review by AAGaming00 — CHANGES_REQUESTED** [review]

File: frontend/src/plugin-loader.tsx (hunk 2 above)
Comment (botatooo) [reply frontend/src/plugin-loader.tsx:106]: The callServerMethod function does indeed need to be awaited. I'll push a commit to fix that whenever I have a chance.

Comment (botatooo) [reply frontend/src/plugin-loader.tsx]: The fix has been pushed

**Comment by AAGaming00** [comment]
I'll review this tomorrow.

**Review by AAGaming00 — APPROVED** [review]
LGTM, please rebase & resolve conflicts.

**Comment by botatooo** [comment]
I did what I could.

**Comment by TrainDoctor** [comment]
Thanks for your work on this PR, apologies we let it fall behind. Supplanted by #128

**Comment by Scrumplex** [comment]
My PR above does not offer a way to reload all plugins in general

**Comment by TrainDoctor** [comment]
> My PR above does not offer a way to reload all plugins in general

While true yours is a better implementation.

**Comment by botatooo** [comment]
This PR wasn't meant to reload plugins, but to reload the list to account for plugins that were added into the plugin folder and similar that SDH didn't initialize because they weren't present in the folder at the time of initialization.

**Pull request closed without merging**